google-home-community.groovy

```diff
@@ -2091,6 +2443,46 @@ private deviceStateForTrait_Dock(deviceTrait, device) {
     ]
 }
 
+@SuppressWarnings('UnusedPrivateMethod')
+private deviceStateForTrait_EnergyStorage(deviceTrait, device) {
+    def deviceState = [:]
+    descriptiveCapacityRemaining = device.currentValue(deviceTrait.descriptiveCapacityRemainingAttribute)
+    if (descriptiveCapacityRemaining == null) {
+        descriptiveCapacityRemaining = "HIGH"
+    }
+    deviceState.descriptiveCapacityRemaining =  descriptiveCapacityRemaining
+    deviceState.capacityRemaining = [
+        [
+            rawValue: device.currentValue(deviceTrait.capacityRemainingRawValue).toInteger(),
+            unit:     deviceTrait.capacityRemainingUnit,
+        ]
+    ]
+    capacityUntilFullRawValue = device.currentValue(deviceTrait.capacityUntilFullRawValue).toInteger()
```

Comment:
This line still throws a `NullPointerException` when `capacityUntilFullRawValue` is unset.  This should leave the `capacityUntilFull` array empty in the returned state in that case.

Reply:
Actually, `capacityUntilFull` should probably just be left unset.  It's not required.  The same is probably true for `isCharging` and `isPluggedIn`.

Reply:
Ahhh.. I was misreading the API document as that was required, but if it's not reported, then it's not.  Let me clean that up.

Reply:
Cleaned up section, and suppressed traits that are not enabled